services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/api_client/_base.py

```diff
@@ -0,0 +1,182 @@
+import asyncio
+import functools
+import inspect
+import logging
+from logging import Logger
+from typing import Any, Awaitable, Callable, Optional
+
+from httpx import AsyncClient, ConnectError, HTTPError, PoolTimeout, Response
+from httpx._types import TimeoutTypes, URLTypes
+from tenacity import RetryCallState
+from tenacity._asyncio import AsyncRetrying
+from tenacity.before import before_log
+from tenacity.retry import retry_if_exception_type
+from tenacity.stop import stop_after_attempt
+from tenacity.wait import wait_exponential
+
+from ._errors import (
+    ClientHttpError,
+    UnexpectedStatusError,
+    _RetryRequestError,
+    _WrongReturnType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _log_requests_in_pool(client: AsyncClient, event_name: str) -> None:
```

Comment:
MINOR: ``"WHY SO LOAD THIS WARNING"``  ? :-)

TIP: Generally, it is more flexible to compose the message and let the user decide how to log later. So it would be like
 
```python
def log_args_from_requests_in_pool(client: AsyncClient, event_name: str):
  return ( "Requests while event '%s': %s",
        event_name.upper(),
        [
            (r.request.method, r.request.url, r.request.headers)
            for r in client._transport._pool._requests
        ],
   )

# now it can be used as
logger.info( **log_args_from_requests_in_pool(client, event_name ) )
logger.warning( **log_args_from_requests_in_pool(client, event_name ) )
logger.error( **log_args_from_requests_in_pool(client, event_name ) )

```

Reply:
Will reduce caps usage, will keep as is since I think it should be logged as warning in both cases.

Reply:
OK. 
On the one hand you enforce warning 👍 , on the other you need to learn about yet another indirection 👎